compiler/p/codegen/OMRInstOpCodeProperties.hpp

```diff
@@ -196,7 +196,7 @@
    /* .name        = */ "addic",
    /* .description =    "Add immediate carrying", */
    /* .opcode      = */ 0x30000000,
-   /* .format      = */ FORMAT_RT_RA_SI16,
+   /* .format      = */ FORMAT_RT_D16_RA,
    /* .minimumALS  = */ TR_Processor::TR_PPCrios1,
```

Comment:
this one should have a different format from addi which has unique gr0 implication.

Reply:
I looked into this and you're right that this is incorrect. However, this brings up an interesting conundrum: we already have code that indirectly assumes that using this as a `Trg1Mem` instruction is correct by using `addi2` [1], which can be transformed into an `addic.` by peepholes. Making this format correct would cause asserts at those locations if the relevant peephole kicks in, so this isn't a trivial fix. Perhaps I should open an issue to track this mismatch in expectations?

[1] https://github.com/eclipse/omr/blob/1d3980b61ba7ef43d0f113cf9f2ad88f79d885ab/compiler/p/codegen/OMRTreeEvaluator.cpp#L1287

Reply:
addi2 was created for having record-form exactly for peephole (and instruction scheduler --- knowing CA bit dependency) to work.  The gr0 restriction on addi (and addi2) is a good subset for addic (instead of bad superset).  Could you clarify the extra asserts?

Reply:
The assert that would fire if the format were changed would be firing due to an unhandled format in `TR::PPCTrg1MemInstruction::generateBinaryEncoding`. That function only has handling code for formats that are documented to imply the `gr0` restriction, i.e. the `MEM`-suffixed formats and formats using `D16`.

Since `TR::MemoryReference` does have some internal logic that explicitly assumes that the `gr0` restriction holds (though I don't know whether it's possibly to make that logic kick in for the particular way we generate `addi2` memory instructions), that's the distinction I used in the documentation to describe the difference between these formats. However, the only difference between `FORMAT_RT_RA_SI16` and `FORMAT_RT_D16_RA` to the binary encoder in practice is whether or not the handling code for `Trg1Mem` instructions will be allowed to kick in. In all other respects, the binary encoder treats them exactly the same (similar to the difference between `FORMAT_RT_RA_RB` and `FORMAT_RT_RA_RB_MEM`). Perhaps as a fix, I could override the format of `addi2_r` explicitly (rather than inheriting the format of `addic_r`) to make only that instruction have the memory format, leave a comment in the code outlining this, and open an issue for further discussion about what should be done? That would allow this PR to make progress without completely dropping this.

Reply:
that was my general concern i raised previously: fragmented implementation --- same ISA format (D-form in this case), different implementation with D16_RA and RA_SI16.

I am fine with temporary work-around in order to move forward.

Reply:
Fixed in dc6078d8275fae9656031d56c7efb934efe2505b.